lib/vyasa_web/components/contexts/read.ex

```diff
@@ -477,7 +506,7 @@ defmodule VyasaWeb.Context.Read do
   defp mutate_draft_reflector(
          %{assigns: %{draft_reflector: %Vyasa.Sangh.Sheaf{} = curr_sheaf, marks: marks}} = socket
        ) do
-    {:ok, com} = Vyasa.Sangh.update_sheaf(curr_sheaf, %{marks: marks})
+    {:ok, com} = Vyasa.Sangh.update_sheaf(curr_sheaf, %{marks: marks |> Enum.reject(fn m -> m.state == :tomb end)})
 
     socket
     |> assign(:draft_reflector, com)
```

Comment:
could this function do a "always keep order values contiguous"?

Reply:
I attempted this but wasn't successful because I still have collisions. 



```elixir
  # Helper function that syncs and mutates Draft Reflector
  defp mutate_draft_reflector(
         %{assigns: %{draft_reflector: %Vyasa.Sangh.Sheaf{} = curr_sheaf, marks: marks}} = socket
       ) do
    sanitised_marks =
      marks
      |> Enum.reverse()
      |> Enum.reject(fn m -> m.state == :tomb end)
      |> Enum.with_index(1)
      |> Enum.map(fn {mark, index} -> %Mark{mark | order: index} end)

    {:ok, com} =
      Vyasa.Sangh.update_sheaf(curr_sheaf, %{
        marks: sanitised_marks
      })

    socket
    |> assign(:draft_reflector, com)
  end
```

see here, the third guy should be coming before the fourth guy but they both have the same index

![image](https://github.com/user-attachments/assets/8553befd-ee39-447a-94a7-b2c86565a39c)